code/modules/wiremod/components/action/pathfind.dm

```diff
@@ -0,0 +1,124 @@
+/**
+ * # Pathfinding component
+ *
+ * Calcualtes a path, returns a list of entities. Each entity is the next step in the path. Can be used with the direction component to move.
+ */
+/obj/item/circuit_component/pathfind
+	display_name = "Pathfinder"
+	display_desc = "When triggered, the next step to the target's location as an entity. This can be used with the direction component and the drone shell to make it move on its own. The Id Card input port is for considering ID access when pathing, it does not give the shell actual access."
+	circuit_flags = CIRCUIT_FLAG_INPUT_SIGNAL|CIRCUIT_FLAG_OUTPUT_SIGNAL
+
+	var/datum/port/input/input_X
+	var/datum/port/input/input_Y
+	var/datum/port/input/id_card
```

Comment:
I think this'd be better off as a separate component because you can technically mark the AA card and use that against this.

You could have a new datatype for it too, the access datatype.

Reply:
If you do end up implementing that new datatype, I could add it to the arrest consoles too so that you don't have to log in.